canu/inventory/plays/aruba-aoscx.yml

```diff
@@ -83,26 +81,99 @@
           when: _result.json.state == "error"
           failed_when: _result.json.state == "error"
 
-        # Begin munging operations
-        # Per the aruba docs, it's not possible to have two checkpoints with the same configuration
-        # Thus, a checkpoint with this exact JSON cannot be used as a new checkpoint or it is detected as a duplicate
-        # As a workaround, one semi-harmless key is modified--the banner
-        # Changing this makes the config different from the running config without any operational side-effects
-        - name: begin munging by setting facts using the existing config
+        - name: get generated config in json format
           ansible.builtin.set_fact:
             original_config: "{{ _result.json.configs.json }}"
 
-        # # TODO: Technically, we should do another dry run here to validate there is no garbage in the config
+        # Get the mgmt interface config from the running-config and apply it to the generated config.
+        # This should prevent users from accidentally locking themselves out.
+        # We can't get query the API direclty for the mgmt interface config because
+        # there are additional fields added that won't work when we upload the running-config json payload.
+        - name: get mgmt int config from running config
+          ansible.builtin.uri:
+            url: "{{ running_config_url }}"
+            validate_certs: false
+            use_proxy: false
+            method: GET
+            return_content: true
+            headers:
+              Cookie: "{{ login.set_cookie }}"
+          register: running_config_data
+
+        - name: get mgmt int json
+          ansible.builtin.set_fact:
+            mgmt_int: "{{ running_config_data.json.System.mgmt_intf }}"
+
+        - name: add mgmt int config to generated config
+          ansible.utils.update_fact:
+            updates:
+              - path: original_config.System.mgmt_intf
+                value: "{{ mgmt_int }}"
+          register: updated_config
+
         - name: write jsonified config to a local file
           ansible.builtin.copy:
-            content: "{{ original_config | to_json(indent=4) }}"
+            content: "{{ updated_config.original_config | to_json(indent=4) }}"
             dest: "{{ validated_config_json }}"
 
+        # set checkpoint for 1 minute.
```

Comment:
Is 1 minute sufficient amount of time for the automation and being able to log in and disable the auto rollback?

Reply:
For my testing it was good (3 switches).  Once we start testing this on bigger internal systems we can adjust accordingly.